Tests/Extractor/JsExtractorTest.php

```diff
@@ -0,0 +1,149 @@
+use Symfony\Component\Translation\MessageCatalogue;
+use Bazinga\Bundle\JsTranslationBundle\Finder\FinderFactory;
+use Bazinga\Bundle\JsTranslationBundle\Filesystem\Filesystem;
+use Bazinga\Bundle\JsTranslationBundle\Extractor\JsExtractor;
+
+final class JsExtractorTest extends PHPUnit_Framework_TestCase
+{
+    const TEST_LOCALE = 'en';
+    const TEST_KEY_1 = 'test-key-1';
+    const TRANSLATION_PATH_PUBLIC = '/translation-path/public';
+    const TRANSLATION_PATH_VIEWS = '/translation-path/views';
+
+    /**
+     * @var JsExtractor
+     */
+    private $sut;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var FinderFactory
+     */
+    private $finderFactory;
+
+    /**
+     * @var MessageCatalogue
+     */
+    private $messageCatalogue;
+
+    private $folder;
+    private $fileName;
+    private $fileContent;
+
+    public function setUp()
+    {
+        $this->filesystem = $this->prophesize(Filesystem::class);
+        $this->finderFactory = $this->prophesize(FinderFactory::class);
+        $this->messageCatalogue = new MessageCatalogue(self::TEST_LOCALE);
+
+        $this->sut = new JsExtractor(
+            $this->filesystem->reveal(),
+            $this->finderFactory->reveal()
+        );
+    }
+
+    public function testExtractShouldRetrieveTransKey()
+    {
+        $this->givenASourceFolderWithATransFunctionUsage();
+        $this->thenTheFinderWillFindAJsFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheTransKeyIsInMessageCatalogue();
+    }
+
+    public function testExtractShouldRetrieveTransChoiceKey()
+    {
+        $this->givenASourceFolderWithATransChoiceFunctionUsage();
+        $this->thenTheFinderWillFindAJsFile();
+        $this->andTheFilesystemWillGrabItsContent();
+        $this->whenUsingTheExtractFromTheSut();
+        $this->assertTheTransChoiceKeyIsInMessageCatalogue();
+    }
+
+    private function givenASourceFolderWithATransFunctionUsage()
+    {
+        $this->givenASourceFolder();
+
+        $this->fileContent = <<<STRING
+        Translator.trans('test-key-1');
+        Translator.trans();
+        Translator.trans(variable);
+STRING;
+    }
+
+    private function givenASourceFolderWithATransChoiceFunctionUsage()
+    {
+        $this->givenASourceFolder();
+
+        $this->fileContent = <<<STRING
+        Translator.transChoice('test-key-1', 5);
+        Translator.transChoice();
+        Translator.transChoice(variable, 5);
+STRING;
+    }
+
+    private function givenASourceFolder()
+    {
+        $this->folder = self::TRANSLATION_PATH_VIEWS;
+        $this->fileName = 'test.js';
+        $this->filesystem
+            ->exists(self::TRANSLATION_PATH_PUBLIC)
+            ->willReturn(true);
+    }
+
+    private function thenTheFinderWillFindAJsFile()
+    {
+        $finder = $this->prophesize(Finder::class);
+
+        $finder
+            ->files()
```

Comment:
But there is **nothing** in your test ensuring that the way you configured the finder is actually the right one to find the files you want to find. so your test currently says "if the code calls the Finder in the way it currently calls it, things are working". This is wrong. I does not actually ensure that configuring the Finder this way is what makes things work.
And it does not protect you from changes in the Finder behavior either.

What you need here is an integration test, which could use the real Finder. You should not mock the finder as it is outside your domain boundary. Using a unit test to cover this code is precisely the issue.